src/App.tsx

```diff
@@ -15,71 +16,84 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export class App extends React.Component {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  handlerSelect = (good: string) => {
+    this.setState({ selectedGood: good });
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handlerClear = () => {
+    this.setState({
+      selectedGood: '',
+    });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  helperSelector = (good: string): (() => void) | undefined => {
+    if (this.state.selectedGood === good) {
+      return this.handlerClear;
+    }
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return () => this.handlerSelect(good);
+  }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good">
-          <td>
+    return (
+      <main className="section container">
+        {!selectedGood ? (
+          <h1 className="title">No goods selected</h1>
+        ) : (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedGood} is selected`}
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
-              data-cy="AddButton"
+              onClick={this.handlerClear}
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              className="delete ml-3"
+            />
+          </h1>
+        )}
+
+        <table className="table">
+          <tbody>
+            {goods.map((good: string) => (
+              <tr
+                data-cy="Good"
+                className={cn({
+                  'has-background-success-light': selectedGood === good,
+                })}
+              >
+                <td>
+                  <button
+                    onClick={this.helperSelector(good)}
```

Comment:
in onClick atribute you need to pass function, not a result of function its can be callBack, if you need to pass some args, 
exm.: {() => yourFunc(args)}